crates/weaver_semconv/src/semconv.rs

```diff
@@ -13,11 +14,82 @@ use std::path::Path;
+    /// Version 2 of the semantic convention schema.
+    #[serde(rename = "2")]
+    V2(SemConvSpecV2),
+}
+
+// Note: We automatically create the Schemars code and provide `allow(unused_qualifications)` to work around schemars limitations.
+// You can use `cargo expand -p weaver_semconv` to find this code and generate it in the future.
+const _: () = {
+    #[automatically_derived]
+    #[allow(unused_braces, unused_qualifications)]
+    impl schemars::JsonSchema for Versioned {
+        fn schema_name() -> std::string::String {
+            "Versioned".to_owned()
+        }
+        fn schema_id() -> std::borrow::Cow<'static, str> {
+            std::borrow::Cow::Borrowed("weaver_semconv::semconv::Versioned")
+        }
+        fn json_schema(generator: &mut schemars::gen::SchemaGenerator) -> schemars::schema::Schema {
+            schemars::_private::metadata::add_description(
+                schemars::schema::Schema::Object(schemars::schema::SchemaObject {
+                    subschemas: Some(Box::new(schemars::schema::SubschemaValidation {
+                        one_of: Some(<[_]>::into_vec(Box::new([
+                            schemars::_private::metadata::add_description(
+                                schemars::_private::new_internally_tagged_enum(
+                                    "version", "1", false,
+                                ),
+                                "Version 1 of the semantic convention schema.",
+                            )
+                            .flatten(
+                                <SemConvSpecV1 as schemars::JsonSchema>::json_schema(generator),
+                            ),
+                            schemars::_private::metadata::add_description(
+                                schemars::_private::new_internally_tagged_enum(
+                                    "version", "2", false,
+                                ),
+                                "Version 2 of the semantic convention schema.",
+                            )
+                            .flatten(
+                                <SemConvSpecV2 as schemars::JsonSchema>::json_schema(generator),
+                            ),
+                        ]))),
+                        ..Default::default()
+                    })),
+                    ..Default::default()
+                }),
+                "A versioned semantic convention file.",
+            )
+        }
+    }
+};
+
+/// A semantic convention file as defined [here](https://github.com/open-telemetry/build-tools/blob/main/semantic-conventions/syntax.md)
+/// A semconv file is a collection of semantic convention groups (i.e. [`GroupSpec`]).
+#[derive(Serialize, Deserialize, Debug, Clone, JsonSchema, PartialEq)]
 #[serde(deny_unknown_fields)]
-pub struct SemConvSpec {
+pub struct SemConvSpecV1 {
     /// A collection of semantic convention groups or [`GroupSpec`].
+    #[serde(default)] // TODO - Should we have disjoint V1 and V2 instead?
```

Comment:
I don’t understand this todo, since we are in the context of SemConvSpecV1.

Reply:
AH, I had fixed this todo but forgot to remove it...